### PR TITLE
feat: Update controller runtime

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this PR do?
Update controller runtime to v0.9.5.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20300

### Depends on:
https://github.com/eclipse/che/issues/20425 with pull request: https://github.com/che-incubator/kubernetes-image-puller-operator/pull/69

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by AndrienkoAleksandr** [comment]
In general pull request is ready for review. I will work on dependencies list in the go.mod, go.sum. Code part I think will be unchanged.

File: pkg/deploy/consolelink_test.go [hunk 1]
```diff
@@ -104,10 +105,7 @@ func TestReconcileConsoleLink(t *testing.T) {
 	// check finalizer
 	c = &orgv1.CheCluster{}
 	err = cli.Get(context.TODO(), types.NamespacedName{Namespace: "eclipse-che", Name: "eclipse-che"}, c)
-	if err != nil {
+	if !errors.IsNotFound(err) {
 		t.Fatalf("Failed to get checluster: %v", err)
 	}
-	if util.ContainsString(c.ObjectMeta.Finalizers, ConsoleLinkFinalizerName) {
```
Comment (tolusha) [inline pkg/deploy/consolelink_test.go:110]: Why did you remove that?

File: pkg/deploy/kubernetes_image_puller.go [hunk 2]
```diff
@@ -276,7 +277,8 @@ func CheckNeededImagePullerApis(ctx *DeployContext) (bool, bool, bool, error) {
 
 	for _, l := range resourcesList {
 		for _, r := range l.APIResources {
-			if l.GroupVersion == chev1alpha1.SchemeGroupVersion.String() && r.Kind == "KubernetesImagePuller" {
+			fmt.Printf("===%s", chev1alpha1.SchemeBuilder.GroupVersion.String())
```
Comment (tolusha) [inline pkg/deploy/kubernetes_image_puller.go:280]: pls remove before merging

File: controllers/devworkspace/sync/sync.go [hunk 3]
```diff
@@ -71,35 +66,24 @@ func (s *Syncer) Sync(ctx context.Context, owner metav1.Object, blueprint metav1
...
-	if err != nil && !errors.IsNotFound(err) {
-		return err
+	if err := s.client.Get(ctx, key, object); err == nil {
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		if err = s.client.Delete(ctx, object); err != nil {
+			return err
+		}
 	}
```
Comment (tolusha) [inline controllers/devworkspace/sync/sync.go:79]: You'e changed a logic, so:
1. declare `var err error`
2. revert `if`clause.

File: controllers/devworkspace/sync/sync.go [hunk 4]
```diff
@@ -91,15 +81,9 @@ func (s *Syncer) Delete(ctx context.Context, object metav1.Object) error {
...
-	kind := blueprintObject.GetObjectKind().GroupVersionKind().Kind
-	if !ok {
-		return nil, fmt.Errorf("object %T is not a runtime.Object. Cannot sync it", blueprint)
-	}
-
-	actual := blueprintObject.DeepCopyObject()
-
+func (s *Syncer) create(ctx context.Context, owner client.Object, key client.ObjectKey, blueprint client.Object) (client.Object, error) {
+	actual := blueprint.DeepCopyObject().(client.Object)
+	kind := blueprint.GetObjectKind()
```
[the remark this answers is not in the document]
Comment (AndrienkoAleksandr) [reply controllers/devworkspace/sync/sync.go:86]: I think "kind" we are using to log object, which will be created. "actual" variable that's copy of the original bluprint object. For multithreding code, I think that's a good idea to use copy of the object... But I think our controller code isn't async, so I guess we can use blueprint object directly. @metlos could you assist, please? Do you agree or maybe I missed something?

[the remark this answers is not in the document]
Comment (tolusha) [reply controllers/devworkspace/sync/sync.go]: I got it. Thank youy

**Review by tolusha — APPROVED** [review]

File: controllers/devworkspace/sync/sync.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (metlos) [reply controllers/devworkspace/sync/sync.go]: The copy is there so that we don't mutate the passed in `blueprint` object, but rather read into its copy, the `actual`. Not sure about the change from `metav1.Object` to `client.Object` but if it all compiles, then I guess we're fine.
